src/mumble/CustomElements.cpp

```diff
@@ -152,6 +153,42 @@ void ChatbarTextEdit::setDefaultText(const QString &new_default, bool force) {
 	}
 }
 
+void ChatbarTextEdit::insertFromMimeData(const QMimeData *source) {
+	if (source->hasImage()) {
+		if (g.bAllowHTML) {
+			QImage image = qvariant_cast<QImage>(source->imageData());
+
+			QString imgHtml = QLatin1String("<br />") + Log::imageToImg(image);
+
+			if (static_cast<unsigned int>(imgHtml.length()) < g.uiImageLength) {
+				emit pastedImage(imgHtml);
+			} else {
+				g.l->log(Log::Information, tr("Unable to send image %1: too large.").arg(QLatin1String("")));
+			}
+		}
+	} else if (source->hasUrls()) {
+		QList<QUrl> urlList = source->urls();
```

Comment:
Maybe a comment should be added explaining what this branch is doing. 'cause when I read "URL", I don't think about paths to local files (actually I think in these cases the proper term would be URI anyways)...